### PR TITLE
Fix video is not started if an operator makes a request twice

### PR DESCRIPTION
**Pull request by DavDo**
MOB - 1453

**Jira issue:**
https://glia.atlassian.net/browse/MOB-1453

**Additional info:**
Before these changes, we were keeping only one instance of `DialogsState` and were able to show only one dialog at a time, which means in some cases we lost important information or had a broken logic e.g. when some dialog was shown and the second was arriving the same time we just kept the first one and lost the second that's why the video or audio was not started if an operator makes a request twice (we were answering to the old request) and so on. Also, I changed a lot of implementations of DialogState to @intDef which is faster. 

**Screenshots:**

### CODE REVIEW

File: widgetssdk/src/main/java/com/glia/widgets/core/dialog/DialogManager.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        stateDeque = new LinkedBlockingDeque<>();
+    }
+
+    public void offer(DialogState state) {
+        if (isDialogShown()) {
+            emitNone();
+        }
+
+        boolean removeLastoccurence = stateDeque.removeLastOccurrence(state);
+        boolean offer = stateDeque.offer(state);
```
Comment (andrews-moc) [inline widgetssdk/src/main/java/com/glia/widgets/core/dialog/DialogManager.java:27]: Looks like `removeLastoccurence` and `offer` variables aren't used, are they?
And minor catch - `removeLastoccurence` => `removeLastOccurence`

Comment (DavDo) [reply widgetssdk/src/main/java/com/glia/widgets/core/dialog/DialogManager.java:27]: > Looks like `removeLastoccurence` and `offer` variables aren't used, are they?
> And minor catch - `removeLastoccurence` => `removeLastOccurence`

You're right, I added variables for test purposes and forgot to delete them.

**Comment by andrews-moc** [comment]
@DavDo Could you also please describe the general idea behind these changes? It will help to understand and review the changes better.

**Comment by DavDo** [comment]
> @DavDo Could you also please describe the general idea behind these changes? It will help to understand and review the changes better.

Updated` Additional Info:` section

File: widgetssdk/src/main/java/com/glia/widgets/chat/ChatView.java [hunk 2]
```diff
@@ -571,34 +571,41 @@ private void updateNewMessageOperatorStatusView(String operatorProfileImgUrl) {
...
-                post(this::showNoMoreOperatorsAvailableDialog);
-            } else if (dialogsState instanceof DialogsState.EngagementEndedDialog) {
-                post(this::showEngagementEndedDialog);
-            } else if (dialogsState instanceof DialogsState.StartScreenSharingDialog) {
-                post(this::showScreenSharingDialog);
-            } else if (dialogsState instanceof DialogsState.EnableNotificationChannelDialog) {
-                post(this::showAllowNotificationsDialog);
-            } else if (dialogsState instanceof DialogsState.EnableScreenSharingNotificationsAndStartSharingDialog) {
-                post(this::showAllowScreenSharingNotificationsAndStartSharingDialog);
+        dialogCallback = dialogState -> {
```
Comment (AndriiHorishniiMOC) [inline widgetssdk/src/main/java/com/glia/widgets/chat/ChatView.java:574]: I don't like those code duplications in ChatView and CallView. Maybe we should refactor it in the future.

Comment (DavDo) [reply widgetssdk/src/main/java/com/glia/widgets/chat/ChatView.java:574]: > I don't like those code duplications in ChatView and CallView. Maybe we should refactor it in the future.

I'll try to refactor tomorrow.

**Review by andrews-moc — APPROVED** [review]

**Review by gugalo — APPROVED** [review]

**Pull request merged**